modules/creating-cluster-with-aws-kms-key.adoc

```diff
@@ -0,0 +1,168 @@
+                  "AWS": "arn:aws:iam::<aws_account_id>:role/ManagedOpenShift-HCP-ROSA-Installer-Role"
+              },
+              "Action": [
+                  "kms:CreateGrant",
+                  "kms:DescribeKey",
+                  "kms:GenerateDataKeyWithoutPlaintext"
+              ],
+              "Resource": "*"
+          },
+          {
+              "Sid": "ROSA KubeControllerManager Permissions",
+              "Effect": "Allow",
+              "Principal": {
+                  "AWS": "arn:aws:iam::<aws_account_id>:role/<operator_role_prefix>-kube-system-kube-controller-manager"
+
+              },
+              "Action": "kms:DescribeKey",
+              "Resource": "*"
+          },
+          {
+              "Sid": "ROSA KMS Provider Permissions",
+              "Effect": "Allow",
+              "Principal": {
+                  "AWS": "arn:aws:iam::<aws_account_id>:role/<operator_role_prefix>-kube-system-kms-provider"
+              },
+              "Action": [
+                  "kms:Encrypt",
+                  "kms:Decrypt",
+                  "kms:DescribeKey"
+              ],
+              "Resource": "*"
+          },
+          {
+              "Sid": "ROSA NodeManager Permissions",
+              "Effect": "Allow",
+              "Principal": {
+                  "AWS": "arn:aws:iam::<aws_account_id>:role/<operator_role_prefix>-kube-system-capa-controller-manager"
+              },
+              "Action": [
+                  "kms:DescribeKey",
+                  "kms:GenerateDataKeyWithoutPlaintext",
+                  "kms:CreateGrant"
+              ],
+              "Resource": "*"
+          }
+      ]
+  } 
+----
+
+. Confirm the details of the policy file created by running the following command:
++
+[source,terminal]
+----
+$ cat rosa-key-policy.json
+----
+
+. Apply the newly generated key policy to the custom KMS key by running the following command:
+
++
+[source,terminal]
+----
+$ aws kms put-key-policy --key-id $KMS_ARN \
+--policy file://rosa-key-policy.json \
+--policy-name default
+----
+
+. Create the cluster by running the following command:
+
++
+[source,terminal]
+----
+$ rosa create cluster --cluster-name <cluster-name> \
+--subnet-ids <private-subnet-id>,<public-subnet-id> \
+--sts \
+--mode auto \
+--machine-cidr 10.0.0.0/16 \
+--compute-machine-type m5.xlarge \
+--hosted-cp \
+--region <aws-region> \
+--oidc-config-id $OIDC_ID \
+--kms-key-arn $KMS_ARN \ <1>
+--etcd-encryption-kms-arn $KMS_ARN \ <2>
+--operator-roles-prefix $OPERATOR_ROLES_PREFIX
+----
+<1> This KMS key ARN is used to encrypt all worker node root volumes. It is not required if only etcd database encryption is needed.
+<2> This KMS key ARN is used to encrypt the etcd database. The etcd database is always encrypted by default with an AES cipher block, but can be encrypted instead with a KMS key. It is not required if only node root volume encryption is needed.
+
+. To verify if the KMS key works, follow these steps: 
```

Comment:
Not critical, but this could be moved to a `.Verification` section

Reply:
complete